### PR TITLE
Add a new func `ReadAtMost` and rename the file

### PR DESCRIPTION
**Pull request by haiyanmeng**
Add a new func `ReadAtMost` and rename the file to a more general name

I planned to add `ReadAtMost` into `k8s.io/pkg/util/system` (see https://github.com/kubernetes/kubernetes/pull/76518), and @tallclair  [suggested](https://github.com/kubernetes/kubernetes/pull/76518#discussion_r277087594) that `k8s.io/utils/io` is a btter place for this func.

### CODE REVIEW

File: io/read.go [hunk 1]
```diff
@@ -53,3 +55,14 @@ func consistentReadSync(filename string, attempts int, sync func(int)) ([]byte,
...
 	return nil, fmt.Errorf("could not get consistent content of %s after %d attempts", filename, attempts)
 }
+
+// ReadAtMost reads up to `limit` bytes from `r`, and reports an error
+// when `limit` bytes are read.
+func ReadAtMost(r io.Reader, limit int64) ([]byte, error) {
+	limitedReader := &io.LimitedReader{R: r, N: limit}
+	data, err := ioutil.ReadAll(limitedReader)
+	if limitedReader.N <= 0 {
+		return data, errors.New("the read limit is reached")
```
Comment (tallclair) [inline io/read.go]: I think the error should be defined as an exported global - see https://github.com/kubernetes/kubernetes/pull/76518#discussion_r277083231

File: io/read.go [hunk 2]
```diff
@@ -53,3 +55,14 @@ func consistentReadSync(filename string, attempts int, sync func(int)) ([]byte,
 	}
 	return nil, fmt.Errorf("could not get consistent content of %s after %d attempts", filename, attempts)
 }
+
+// ReadAtMost reads up to `limit` bytes from `r`, and reports an error
+// when `limit` bytes are read.
+func ReadAtMost(r io.Reader, limit int64) ([]byte, error) {
+	limitedReader := &io.LimitedReader{R: r, N: limit}
+	data, err := ioutil.ReadAll(limitedReader)
```
Comment (tallclair) [inline io/read.go]: I think you should check `err` before checking the read limit.

File: io/read.go (hunk 1 above)
Comment (haiyanmeng) [reply io/read.go]: Fixed! Thanks!

File: io/read.go (hunk 2 above)
Comment (haiyanmeng) [reply io/read.go]: Done!

**Pull request merged**